src/fides/api/service/connectors/s3_connector.py

```diff
@@ -0,0 +1,75 @@
+from typing import Any, Dict, List, Optional
+
+from loguru import logger
+
+from fides.api.graph.execution import ExecutionNode
+from fides.api.models.connectionconfig import ConnectionTestStatus
+from fides.api.models.policy import Policy
+from fides.api.models.privacy_request import PrivacyRequest, RequestTask
+from fides.api.schemas.connection_configuration.connection_secrets_s3 import S3Schema
+from fides.api.service.connectors.base_connector import BaseConnector
+from fides.api.service.connectors.query_config import QueryConfig
+from fides.api.util.aws_util import get_aws_session
+from fides.api.util.collection_util import Row
+from fides.connectors.models import ConnectorFailureException
+
+
+class S3Connector(BaseConnector):
+    """
+    AWS S3 Connector - this is currently used just to test connections to S3.
+
+    NOTE: No DSR processing is yet supported for S3.
+    """
+
+    def create_client(self) -> Any:  # type: ignore
+        """Returns a client for s3"""
+        config = S3Schema(**self.configuration.secrets or {})
+        return get_aws_session(
+            auth_method=config.auth_method.value,
+            storage_secrets=config.dict(),  # type: ignore
+            assume_role_arn=config.aws_assume_role_arn,
+        )
+
+    def query_config(self, node: ExecutionNode) -> QueryConfig[Any]:
+        """DSR execution not yet supported for S3"""
+        raise NotImplementedError()
+
+    def test_connection(self) -> Optional[ConnectionTestStatus]:
+        """
+        Connects to AWS S3 and lists buckets to validate credentials.
+        """
+        logger.info("Starting test connection to {}", self.configuration.key)
+        try:
+            session = self.client()
+            s3 = session.resource("s3")
+            s3.buckets.all()
+        except Exception as error:
+            raise ConnectorFailureException(str(error))
```

Comment:
This is logging sensitive credentials to the terminal

Reply:
😱 

clearly need to address that.

Reply:
so i'm not seeing this now, with my most recent changes. frankly i hadn't seen it before (just because i hadn't been looking closely) - but could you let me know specifically what you were doing to cause the sensitive credentials to be logged, so that we can be sure it's no longer a problem?

Reply:
I believe I was just using the user arn from your original credentials which was causing that to get logged, but i wasn't seeing the access ikd/key.  I'm not seeing this anymore, it looks good.